### PR TITLE
[Spree Upgrade] Improve failed_checkout_spec by making calculator decorator line_items_for work with new Stock::Package

### LINKED ISSUE #3025: [Spree Upgrade] Make calculator_decorator.line_items_for work with Stock::Package (1 broken test)
Similarly to what we did we the weight calculator in #2931, we need to make the calculator_decorator.line_items_for work with Stock::Package.

This will improve the spec spec/requests/checkout/failed_checkout_spec.rb:42.

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Closes #3025 

#### What should we test?
failed_checkout_spec should be failing on a new error.

### CODE REVIEW

File: spec/models/spree/calculator/flat_percent_item_total_spec.rb [hunk 1]
```diff
@@ -4,13 +4,20 @@
   let(:calculator) { Spree::Calculator::FlatPercentItemTotal.new }
   let(:line_item) { instance_double(Spree::LineItem, amount: 10) }
 
-  before { calculator.stub :preferred_flat_percent => 10 }
+  before { calculator.stub preferred_flat_percent: 10 }
 
-  it "should compute amount correctly for a single line item" do
+  it "computes amount correctly for a single line item" do
```
Comment (sauloperez) [inline spec/models/spree/calculator/flat_percent_item_total_spec.rb]: thanks! I prefer examples this way. `should` doesn't inspire confidence :trollface:

File: spec/models/spree/calculator/flat_percent_item_total_spec.rb [hunk 2]
```diff
@@ -4,13 +4,20 @@
...
 
   context "extends LocalizedNumber" do
     it_behaves_like "a model using the LocalizedNumber module", [:preferred_flat_percent]
   end
+
+  it "computes amount correctly for a given Spree::Stock::Package" do
+    order = double(:order, line_items: [line_item] )
+    package = double(:package, order: order)
+
+    calculator.compute(package).should == 1.0
```
Comment (sauloperez) [inline spec/models/spree/calculator/flat_percent_item_total_spec.rb]: Could we use RSpec-3 syntax? or did we decide to stick to Rspec-2 in the upgrade? I don't remember.

File: spec/models/spree/calculator/flat_percent_item_total_spec.rb [hunk 3]
```diff
@@ -4,13 +4,20 @@
...
-  it "should compute amount correctly for a single line item" do
+  it "computes amount correctly for a single line item" do
     calculator.compute(line_item).should == 1.0
   end
 
   context "extends LocalizedNumber" do
     it_behaves_like "a model using the LocalizedNumber module", [:preferred_flat_percent]
   end
+
+  it "computes amount correctly for a given Spree::Stock::Package" do
```
Comment (sauloperez) [inline spec/models/spree/calculator/flat_percent_item_total_spec.rb]: Should it mention [Stock::Package](https://github.com/openfoodfoundation/openfoodnetwork/blob/2-0-stable/app/models/stock/package.rb) instead?

File: spec/models/spree/calculator/flat_percent_item_total_spec.rb (hunk 2 above)
Comment (luisramos0) [reply spec/models/spree/calculator/flat_percent_item_total_spec.rb]: I tried it in previous merges and it created merge hell. Let's leave that for v2 please. We can go for a mass transpec then.

File: spec/models/spree/calculator/flat_percent_item_total_spec.rb (hunk 3 above)
Comment (luisramos0) [reply spec/models/spree/calculator/flat_percent_item_total_spec.rb]: yes! Fixed. Thanks!

**Comment by luisramos0** [comment]
@mkllnk yes, I considerer that, I really wanted to clean it up BUT I am not sure about it.
There must be a reason why this code was copy pasted into Weight Calculator and I am not sure it's just the "raise "Unknown object" part of it. It's just more work I didn't want to get into now...

**Comment by luisramos0** [comment]
I think I am afraid this was done for some class loading issue as we had in Payment/gateway structure...
Because Weight < Spree::Calculator and this method is in our calculator_decorator...

**Pull request merged**